File: src/main/java/com/tms/csp/transforms/CompoundListTransform.java
package com.tms.csp.transforms;

import com.tms.csp.ast.Exp;

import javax.annotation.Nonnull;
import java.util.List;

import static com.google.common.base.Preconditions.checkNotNull;

public class CompoundListTransform extends ListTransform {

    private final Transformer[] transformer;

    public CompoundListTransform(@Nonnull Transformer... transformer) {
        checkNotNull(transformer);
        this.transformer = transformer;
    }

    @Nonnull
    @Override
    public List<Exp> transform(@Nonnull List<Exp> listIn) {
        return multiTransformList(listIn, transformer);
    }

}

File: src/main/java/com/tms/csp/transforms/CompoundListTransform.kt
package com.tms.csp.transforms

import com.tms.csp.ast.Exp

import com.google.common.base.Preconditions.checkNotNull

class CompoundListTransform(vararg transformer: Transformer) : ListTransform() {

    private val transformer: Array<out Transformer>

    init {
        checkNotNull<Array<out Transformer>>(transformer)
        this.transformer = transformer
    }

    override fun transform(listIn: List<Exp>): List<Exp> {
        return ListTransform.multiTransformList(listIn, transformer)
    }

}
